File: code/chemical-main/backend/src/main/java/com/at/pojo/YoloSummary.java
package com.at.pojo;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class YoloSummary {
    private Integer currentCount;
    private Integer analysisTime;
    private Integer riskCount;
    private Integer onlineDevices;
    private LocalDateTime lastAnalysisTime;
}

File: code/chemical-main/frontend/src/api/yoloSummary.ts
import request from '@/utils/request'
import type { ApiResult } from '@/types/api'

export interface YoloSummary {
  currentCount: number | null
  analysisTime: number | null
  riskCount: number
  onlineDevices: number
  lastAnalysisTime?: string | null
}

export const reqYoloSummary = () =>
  request.get<null, ApiResult<YoloSummary>>('/analysis/summary')
